cupy/manipulation/add_remove.py

```diff
@@ -13,7 +13,63 @@
 # TODO(okuta): Implement resize
 
 
-# TODO(okuta): Implement trim_zeros
+_first_non_zero_krnl = ReductionKernel(
+    'raw S data, T indices, T len, int64 mode',
+    'T y',
+    '''mode? data[indices] ? indices : len
+           : data[len - 1 - indices] ? len - 1 - indices : T(-1)''',
+    'mode? min(a,b): max(a, b) ',
+    'y = a',
+    'mode ? len : -1',
+    'first_non_zero'
+)
+_first_non_zero_cplx_krnl = ReductionKernel(
```

Comment:
Could you explain why this kernel is needed? Why not use `_first_non_zero_krnl` also for complex inputs?

Reply:
It raises error for complex inputs

Reply:
I can use  _first_non_zero_cplx_krnl for both complex and non-complex inputs but I needed extra memory (cupy.zeros) for non-complex inputs to pass it as imag (as in the 1st commit)